WooCommerce/Classes/ViewRelated/Orders/Simple Payments/Summary/SimplePaymentsSummaryViewModel.swift

```diff
@@ -138,6 +148,11 @@ final class SimplePaymentsSummaryViewModel: ObservableObject {
     ///
     func reloadContent() {
         objectWillChange.send()
+
+        // Tracks if the merchant has added a note.
+        if noteContent.isNotEmpty {
+            analytics.track(event: WooAnalyticsEvent.SimplePayments.simplePaymentsFlowNoteAdded())
```

Comment:
I noticed that this fires in multiple scenarios:

* When I first add an order note
* If I edit the order note but tap the cancel button
* If I edit the order note and save changes (with order content)

Because we only track if a note is added (and not if a note is removed), we also aren't capturing the scenario where a note is added and then removed. (That's probably an unlikely scenario, but wanted to mention it.)

Is that ok for what we plan to use this data for? In particular, the scenario with the "cancel" button seems misleading but may not matter depending on how we use this data.

Reply:
Another nice catch! I didn't want to make this extra complex and ended up creating the unhandled scenarios you mention. 🤦 

Upon review, I have modified this event to be tracked when the user taps the done button in the order note screen, regardless of the content.

I think that should be consistent enough!

Reply:
Fixed in https://github.com/woocommerce/woocommerce-ios/pull/5633/commits/2634f0fea453b734ef19b5134628a7c2026ec3f2